compile errors

Hey hello! I suspect this is a build.gradle issue - I've been working on a fork of mffs for my own personal use - did a few minor edits

Unfortunately when I run gradlew build, it throws a series of errors about necessary packages not being found. I've added the ic2 api to the list of dependencies and its corresponding repository, but I'm also missing libs from "nekokittygames" e.g.

`C:\Users\(myusername)\Documents\Eclipse Projects\modularforcefieldsystem-1.10\build\sources\main\java\com\nekokittygames\mffs\common\item\ItemCardDataLink.java:4: error: package com.nekokittygames.mffs.libs does not exist
import com.nekokittygames.mffs.libs.LibMisc;`

I'm also missing the support in build.gradle for the buildcraft api, but I suspect I can fix that on my own.

--- a/.gitignore
+++ b/.gitignore
@@ -20,4 +20,4 @@ build
 # other
 eclipse
 run
-libs
+/libs

--- a/src/main/java/com/nekokittygames/mffs/common/block/BlockForceField.java
+++ b/src/main/java/com/nekokittygames/mffs/common/block/BlockForceField.java
@@ -19,21 +19,11 @@
  */
 package com.nekokittygames.mffs.common.block;
 
-import buildcraft.api.core.Identifier;
-import com.nekokittygames.mffs.api.IFieldTeleporter;
-import com.nekokittygames.mffs.api.IForceFieldBlock;
-import com.nekokittygames.mffs.api.PointXYZ;
-import com.nekokittygames.mffs.common.*;
-import com.nekokittygames.mffs.common.WorldMap.ForceFieldWorld;
-import com.nekokittygames.mffs.common.item.ItemCardPowerLink;
-import com.nekokittygames.mffs.common.multitool.ItemDebugger;
-import com.nekokittygames.mffs.common.tileentity.TileEntityCapacitor;
-import com.nekokittygames.mffs.common.tileentity.TileEntityForceField;
-import com.nekokittygames.mffs.common.tileentity.TileEntityProjector;
-import com.nekokittygames.mffs.libs.LibBlockNames;
-import com.nekokittygames.mffs.libs.LibMisc;
+import java.util.Random;
+
+import javax.annotation.Nullable;
+
 import net.minecraft.block.Block;
-import net.minecraft.block.BlockContainer;
 import net.minecraft.block.ITileEntityProvider;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.PropertyEnum;
@@ -47,21 +37,41 @@ import net.minecraft.init.Blocks;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.tileentity.TileEntity;
-import net.minecraft.util.*;
+import net.minecraft.util.BlockRenderLayer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumHand;
+import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.math.AxisAlignedBB;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.util.text.TextComponentTranslation;
-import net.minecraft.world.ChunkCache;
 import net.minecraft.world.Explosion;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.ReflectionHelper;
+
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
-import javax.annotation.Nullable;
-import java.util.Random;
+import com.nekokittygames.mffs.api.IFieldTeleporter;
+import com.nekokittygames.mffs.api.IForceFieldBlock;
+import com.nekokittygames.mffs.api.PointXYZ;
+import com.nekokittygames.mffs.common.ForceFieldBlockStack;
+import com.nekokittygames.mffs.common.ForceFieldTyps;
+import com.nekokittygames.mffs.common.Functions;
+import com.nekokittygames.mffs.common.Linkgrid;
+import com.nekokittygames.mffs.common.MFFSDamageSource;
+import com.nekokittygames.mffs.common.ModularForceFieldSystem;
+import com.nekokittygames.mffs.common.SecurityHelper;
+import com.nekokittygames.mffs.common.SecurityRight;
+import com.nekokittygames.mffs.common.WorldMap;
+import com.nekokittygames.mffs.common.WorldMap.ForceFieldWorld;
+import com.nekokittygames.mffs.common.item.ItemCardPowerLink;
+import com.nekokittygames.mffs.common.multitool.ItemDebugger;
+import com.nekokittygames.mffs.common.tileentity.TileEntityCapacitor;
+import com.nekokittygames.mffs.common.tileentity.TileEntityForceField;
+import com.nekokittygames.mffs.common.tileentity.TileEntityProjector;
+import com.nekokittygames.mffs.libs.LibBlockNames;
+import com.nekokittygames.mffs.libs.LibMisc;
 
 public class BlockForceField extends Block implements IForceFieldBlock ,ITileEntityProvider {
 	public static PropertyEnum<ForceFieldTyps> FORCEFIELD_TYPE=PropertyEnum.create("type",ForceFieldTyps.class);

--- a/src/main/java/com/nekokittygames/mffs/common/block/BlockProjector.java
+++ b/src/main/java/com/nekokittygames/mffs/common/block/BlockProjector.java
@@ -66,7 +66,7 @@ public class BlockProjector extends BlockMFFSBase {
 			int type=0;
 			if(proj.get_type()!=null)
 				type=proj.get_type().getForceFieldModuleType();
-			ModularForceFieldSystem.log.info("With state type: "+type);
+			//ModularForceFieldSystem.log.info("With state type: "+type);
 			return super.getActualState(state, worldIn, pos).withProperty(FIELD_TYPE,type);
 		}
 

--- a/src/main/java/com/nekokittygames/mffs/common/tileentity/TileEntityProjector.java
+++ b/src/main/java/com/nekokittygames/mffs/common/tileentity/TileEntityProjector.java
@@ -934,7 +934,7 @@ public class TileEntityProjector extends TileEntityFEPoweredMachine implements
 				&& this.hasOption(
 						ModularForceFieldSystem.MFFSProjectorOptionCamouflage,
 						true))
-			return true;
+			return par1ItemStack.getItem() instanceof ItemBlock;
 
 		if (hasValidTypeMod()) {
 			ModuleBase modTyp = get_type();

--- a/src/main/java/com/nekokittygames/mffs/libs/LibBlockNames.java
+++ b/src/main/java/com/nekokittygames/mffs/libs/LibBlockNames.java
@@ -1,0 +1,36 @@
+/*  
+    Copyright (C) 2012 Thunderdark
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+    
+    Contributors:
+    
+    Matchlighter
+    Thunderdark 
+
+ */
+package com.nekokittygames.mffs.libs;
+
+public class LibBlockNames {
+	public static final String EXTRACTOR = "extractor";
+	public static final String ADV_SECURITY = "advSecurity";
+	public static final String AREA_DEFENSE = "areaDefense";
+	public static final String CAPACITOR = "capacitor";
+	public static final String CONTROL_SYSTEM = "controlSystem";
+	public static final String CONVERTER = "converter";
+	public static final String MONAZIT_ORE = "monazitOre";
+	public static final String PROJECTOR = "projector";
+	public static final String SECURITY_STORAGE = "securityStorage";
+	public static final String FORCE_FIELD = "forceField";
+}

--- a/src/main/java/com/nekokittygames/mffs/libs/LibItemNames.java
+++ b/src/main/java/com/nekokittygames/mffs/libs/LibItemNames.java
@@ -1,0 +1,65 @@
+/*  
+    Copyright (C) 2012 Thunderdark
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+    
+    Contributors:
+    
+    Matchlighter
+    Thunderdark 
+
+ */
+package com.nekokittygames.mffs.libs;
+
+public class LibItemNames {
+	public static final String FORCICIUM = "forcicium";
+	public static final String ACCESS_CARD = "accessCard";
+	public static final String UPGRADE_CAPACITY = "upgradeCapacity";
+	public static final String UPGRADE_RANGE = "upgradeRange";
+	public static final String DATA_LINK_CARD = "dataLinkCard";
+	public static final String EMPTY_CARD = "emptyCard";
+	public static final String PERSONAL_ID = "personalId";
+	public static final String POWER_CARD = "powerCard";
+	public static final String POWER_LINK_CARD = "powerLinkCard";
+	public static final String SECURITY_LINK_CARD = "securityLinkCard";
+	public static final String EXTRACTOR_UPGRADE_BOOSTER = "extractorUpgradeBooster";
+	public static final String FORCE_POWER_CRYSTAL = "forcePowerCrystal";
+	public static final String FORCICIUM_CELL = "forciciumCell";
+	public static final String PROJECTOR_DISTANCE_MODULATOR = "projectorDistance";
+	public static final String PROJECTOR_STENGTH_MODULATOR = "projectorStrength";
+	public static final String PROJECTOR_FOCUS_MATRIX = "projectorFocusMatrix";
+	public static final String MODULE_ADV_CUBE = "moduleAdvCube";
+	public static final String MODULE_CONTAINMENT = "moduleContainment";
+	public static final String MODULE_CUBE = "moduleCube";
+	public static final String MODULE_DEFLECTOR = "moduleDeflector";
+	public static final String MODULE_DIAGONAL_WALL = "moduleDiagonalWall";
+	public static final String MODULE_SPHERE = "moduleSphere";
+	public static final String MODULE_TUBE = "moduleTube";
+	public static final String MODULE_WALL = "moduleWall";
+	public static final String MULTITOOL_DEBUGGER = "multitoolDebugger";
+	public static final String MULTITOOL_TRANSPORTER = "multitoolTransporter";
+	public static final String MULTITOOL_MANUAL = "multitoolManual";
+	public static final String MULTITOOL_ID_WRITER = "multitoolIdWriter";
+	public static final String MULTITOOL_SWITCH = "multitoolSwitch";
+	public static final String MULTITOOL_WRENCH = "multitoolWrench";
+	public static final String OPTION_BLOCK_BREAKER = "optionBlockBreaker";
+	public static final String OPTION_CAMOFLAGE = "optionCamoflage";
+	public static final String OPTION_DEFENSE_STATION = "optionDefenseStation";
+	public static final String OPTION_FIELD_FUSION = "optionFieldFusion";
+	public static final String OPTION_FIELD_MANIPULATOR = "optionFieldManipulator";
+	public static final String OPTION_FIELD_JAMMER = "optionFieldJammer";
+	public static final String OPTION_MOB_DEFENSE = "optionMobDefense";
+	public static final String OPTION_SPONGE = "optionSponge";
+	public static final String OPTION_TOUCH_DAMAGE = "optionTouchDamage";
+}

--- a/src/main/java/com/nekokittygames/mffs/libs/LibMisc.java
+++ b/src/main/java/com/nekokittygames/mffs/libs/LibMisc.java
@@ -1,0 +1,27 @@
+/*  
+    Copyright (C) 2012 Thunderdark
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+    
+    Contributors:
+    
+    Matchlighter
+    Thunderdark 
+
+ */
+package com.nekokittygames.mffs.libs;
+
+public class LibMisc {
+	public static final String UNLOCALIZED_PREFIX = "modularforcefieldsystem.";
+}

